npm/react/src/mount.ts

```diff
@@ -41,6 +41,36 @@ const injectStyles = (options: MountOptions) => {
  **/
 export const mount = (jsx: React.ReactNode, options: MountOptions = {}) => _mount('mount', jsx, options)
 
+const versionRe = /(\d.+?)\./
+
+function getMajorVersion (semver: string) {
+  const match = semver.match(versionRe)
+  const toNum = match?.[1] && parseInt(match?.[1])
+
+  if (!toNum) {
+    throw Error(`Could not extract major version from ${semver}`)
+  }
+
+  return toNum
+}
+
+async function importReactModules () {
+  const react = await import('react')
+  const majorVersion = getMajorVersion(react.version)
+  const reactDomImport = majorVersion <= 17
```

Comment:
Total nitpick: Below we do a `< 18` check, might be good to do that same comparison here just so that it pops up if somebody does a global search for stuff that might be related to React 18

Reply:
Good idea, thanks

Reply:
Made function and using semver: c8fee62